internal/resource/config/authenticationselector/authentication_selector_resource.go

```diff
@@ -0,0 +1,353 @@
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &authenticationSelectorResource{}
+	_ resource.ResourceWithConfigure   = &authenticationSelectorResource{}
+	_ resource.ResourceWithImportState = &authenticationSelectorResource{}
+
+	attributeContractAttrType = map[string]attr.Type{
+		"extended_attributes": types.ListType{ElemType: types.ObjectType{
+			AttrTypes: map[string]attr.Type{
+				"name": types.StringType,
+			}}},
+	}
+)
+
+// AuthenticationSelectorsResource is a helper function to simplify the provider implementation.
+func AuthenticationSelectorsResource() resource.Resource {
+	return &authenticationSelectorResource{}
+}
+
+// authenticationSelectorResource is the resource implementation.
+type authenticationSelectorResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type authenticationSelectorResourceModel struct {
+	AttributeContract   types.Object `tfsdk:"attribute_contract"`
+	SelectorId          types.String `tfsdk:"selector_id"`
+	Id                  types.String `tfsdk:"id"`
+	Name                types.String `tfsdk:"name"`
+	PluginDescriptorRef types.Object `tfsdk:"plugin_descriptor_ref"`
+	ParentRef           types.Object `tfsdk:"parent_ref"`
+	LastModified        types.String `tfsdk:"last_modified"`
+	Configuration       types.Object `tfsdk:"configuration"`
+}
+
+// GetSchema defines the schema for the resource.
+func (r *authenticationSelectorResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages Authentication Selectors",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Description: "The plugin instance name. The name can be modified once the instance is created.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"plugin_descriptor_ref": schema.SingleNestedAttribute{
+				Required:    true,
+				Description: "Reference to the plugin descriptor for this instance. The plugin descriptor cannot be modified once the instance is created.",
+				Attributes:  resourcelink.ToSchemaNoLocation(),
+				PlanModifiers: []planmodifier.Object{
+					objectplanmodifier.RequiresReplace(),
+				},
+			},
+			"parent_ref": schema.SingleNestedAttribute{
+				Description: "The reference to this plugin's parent instance. The parent reference is only accepted if the plugin type supports parent instances.",
+				Optional:    true,
+				Attributes:  resourcelink.ToSchemaNoLocation(),
+			},
+			"configuration": pluginconfiguration.ToSchema(),
+			"last_modified": schema.StringAttribute{
```

Comment:
I think we can leave read-only modification timestamps like this out of the provider. They tend to cause issues with unexpected plans.

Reply:
removed